### DISCUSSION TITLE
How do I set the root directory when using Docker?

### DISCUSSION THREAD

**Discussion by Deejay85** [Difegue/LANraragi#747] [Q&A]
I'm following the above online guide here:

https://sugoi.gitbook.io/lanraragi/installing-lanraragi/docker

So far I'm trying to set the root directory by following the prompt which goes as follows:

`--mount type=bind,source=E:\,target=/home/koyomi/lanraragi/content`

Unfortunately that didn't work as well as I wanted it to. I tried entering in the entire string, then the first half of it followed by the second half, and when that didn't work I tried googling it, which didn't work as well as I wanted it to. Also I have tried to enter this command, both using command prompt, as well as in the terminal section inside of Docker Desktop...neither of which worked. Finally I ran the program, went into settings, and tried adding "/e," which didn't work, and neither did copying and pasting the address from the desktop version of LANraragi (mnt/e).

I've tried just about everything I can think of on my own, and have finally hit a dead end, so can someone please tell me what I am doing wrong so I can fix the situation at hand, and can finally get back to using this thing to finish updating the metadata on my files?

**Comment by Difegue** [comment]
👋 I haven't used Docker Desktop for Windows in a while, but I believe what you want is  
 `--mount type=bind,source=/e,target=/home/koyomi/lanraragi/content` .

See https://docs.docker.com/desktop/troubleshoot/topics/#topics-for-windows , section "Path conversion on Windows"

**Comment by Deejay85** [comment]
Tried entering that in Docker's terminal, and it didn't work. I got back the following error message:

`/bin/sh: --mount: not found`

**Reply by Difegue** [reply]
I've taken another look at the docker doc for windows paths, looks like `e:` is the correct syntax:  
![image](https://user-images.githubusercontent.com/8237712/216313181-d5ad905c-aef2-4256-9e16-e8bf3dbd0c96.png)

Keep in mind you also need to write the bind mount as part of a full `docker` command, so something like:  
```
docker run --name=lanraragi -p 3000:3000 --mount type=bind,source=e:,target=/home/koyomi/lanraragi/content --mount type=bind,source=[YOUR_DATABASE_DIRECTORY],target=/home/koyomi/lanraragi/database difegue/lanraragi
```
(please set the db directory to whatever you're using)
